content/blog/dynamic-providers/index.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "Dynamic Providers"
+date: 2019-12-20
+meta_desc: "Pulumi dynamic providers manage resources that do not have prebuilt providers such as AWS, Azure, and GCP"
+meta_image: meta.png
+authors:
+    - praneet-loke
+tags:
+    - providers
+    - resources
+---
+
+Pulumi has several resource providers (over 45!) that allow you to interact with your favorite cloud provider or resource using a programming language. But what if there isn’t a provider available for what you want to accomplish? That’s where dynamic providers can help you. They can be an extremely important part of your infrastructure app.
+
+<!--more-->
+
+A provider manages the CRUD lifecycle of a resource against some underlying service. The underlying service can be one of the public clouds or an on-premise service. For example, AWS, Azure, GCP, and Kubernetes are all resource providers, which can create/update/destroy resources in the respective clouds.
```

Comment:
Consider adding a section with a distinct title to explain what providers are, maybe add a picture of how Pulumi works

Reply:
Hmm I do have a section below this that is exactly meant for a brief intro to providers. But I didn't go into too much details on purpose. I could link to [this](https://www.pulumi.com/docs/intro/concepts/how-pulumi-works/) page and maybe even include the diagram from there.